Bloom.md

```diff
@@ -0,0 +1,199 @@
+---
+RFC: 4
+Status: Proposed
+---
+
+# ValKeyBloom Module RFC
+
+## Abstract
+
+The proposed feature is ValKeyBloom which is a Rust based Module that brings a native bloom filter data type into Valkey.
+The module should be API-compatible with Redis Ltd.’s ReBloom Module.
+
+## Motivation
+
+Bloom filters are a space efficient probabilistic data structure that can be used to “check” whether an element exists in a set (with a defined false positive), and to “add” elements to a set. While checking whether an item exists, false positives are possible, but false negatives are not possible. https://en.wikipedia.org/wiki/Bloom_filter
+
+To utilize Bloom filters in their client applications, users today use client libraries that are compatible with the ReBloom including jedis, redis-py, node-redis, nredisstack, rueidis, rustis, etc. This allows customers to perform bloom filter based operations, e.g. add and set.
+
+Redis Ltd.‘s ReBloom is published under a proprietery license and hence cannot be distributed freely with ValKey.
+
+There is growing [demand](https://github.com/orgs/valkey-io/discussions?discussions_q=+bloom+) for an (1) Open Source bloom filter feature in Valkey which is (2) compatible with the ReBloom API syntax and with existing ReBloom based client libraries. ValkeyBloom can help address both these requirements.
+
+
+## Design Considerations
+
+The ValkeyBloom module brings in a bloom module data type into Valkey and provides commands to create / reserve bloom filters, operate on them (add items, check if items exist), inspect bloom filters, etc. It allows customization of properties of bloom filter (capacity, false positive rate, expansion rate, specification of scaling vs non-scaling filters etc) through commands and configurations. It also allows users to create scalable bloom filters and back up & restore bloom filters (through RDB load and save).
```

Comment:
This is a description of what's already implemented and how it behaves. I suppose it's meant as a spec for what we want, so I think most of it belongs under Specification rather than Design considerations.

What we want under design consiserations is basically what we have to chose between in the case of different possibilities (like for aof rewrite).